osu.Server.Spectator/Hubs/MultiplayerHub.cs

```diff
@@ -405,6 +412,17 @@ protected virtual async Task EndDatabaseMatch(MultiplayerRoom room)
             }
         }
 
+        protected virtual async Task<bool> CheckUserRestrictions()
```

Comment:
I'd rename to `IsUserRestricted`. With a name like `CheckUserRestrictions` it's not really clear what a `true` value means.